### DISCUSSION TITLE
Disabled, Hostile NPCs still offer quests and services

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14792] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

It is outright strange how NPCs still offer their services or dialog (using triggeraction/conversationaction) if they are:

1. DEEPLY Hostile (<60 reputation, outpost shoots you on sight)
2. Handcuffed

They do not offer services if they are stunned, but can still give out quests if you run at the outpost manager.
Also see https://github.com/FakeFishGames/Barotrauma/discussions/13960

![image](https://github.com/user-attachments/assets/ae950288-31d9-4112-b0de-ef3cb42141df)
![image](https://github.com/user-attachments/assets/c99b8a72-a96d-442b-a864-b2757f161bea)

### Reproduction steps

1. Start campaign
2. setreputation coalition -100
3. Casually grab quests from outpost manager and upgrade your sub

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This is not a bug, it's intentional that you can still interact with them despite your reputation. Even though I see how it can seem weird, I think it's easy to justify as the merchants just being opportunistic enough to sell to anyone who has the money, and it also makes it less likely for the player to get softlocked due to low reputation.

The justification is also given in their dialog lines:
>Honestly, I'd rather not do business with you people, but I don't have the luxury of picking my customers.
>If I wasn't broke, I'd tell you to piss off. But if you've got the money…
>They say I shouldn't be serving you, but I say I shouldn't be turning down a paying customer.

**Reply by ubertpendragon** [reply]
The dialog is understandable if you have low rep (between say -30 and -60) the problem is that if they're OPENLY hostile (below -60) you can snatch quests and items from an outpost that actively seeks to murder you

**Reply by Regalis11** [reply]
I'm not sure I see why the same logic wouldn't apply to openly hostile outposts. Even if the security tries to arrest you and the NPCs try to flee from you, why does that mean merchants would necessarily be unwilling to trade with you, or that the outpost managers would be unwilling to give missions to you?

**Reply by JeroonBosch** [reply]
Personally I think ubertpendragon has a point:
- Generally I imagine the outpost managers are put in charge of the outpost for a reason by the factions, so if you're at a -100 kill on sight reputation with the faction and the outpost manager is actively fighting you, getting missions seems very odd.
- NPCs in active combat state imo shouldn't offer any kind of service, to me fighting and providing service seem at odds
- Same _could_ be said for the fleeing state, although that is more up to debate, we could allow trading (basically at gun point)

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in the next update. Closing discussion.
